src/EventSubscriber/EventSubscriber.php

```
@@ -0,0 +1,105 @@
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Event subscriber.
+ */
+class EventSubscriber implements EventSubscriberInterface {
+  /**
+   * The route match.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  private RouteMatchInterface $routeMatch;
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  private AccountProxyInterface $currentUser;
+
+  /**
+   * The webform helper.
+   *
+   * @var \Drupal\os2forms_rest_api\WebformHelper
+   */
+  private WebformHelper $webformHelper;
+
+  /**
+   * Constructor.
+   */
+  public function __construct(RouteMatchInterface $routeMatch, AccountProxyInterface $currentUser, WebformHelper $webformHelper) {
+    $this->routeMatch = $routeMatch;
+    $this->currentUser = $currentUser;
+    $this->webformHelper = $webformHelper;
+  }
+
+  /**
+   * On request handler.
+   *
+   * Check for user access to webform API resource.
+   */
+  public function onRequest(KernelEvent $event): void {
+    $routeName = $this->routeMatch->getRouteName();
+    $restRouteNames = [
+      'rest.webform_rest_elements.GET',
+      'rest.webform_rest_fields.GET',
+      'rest.webform_rest_submission.GET',
+      'rest.webform_rest_submission.PATCH',
+      'rest.webform_rest_submit.POST',
+    ];
+    if ($this->currentUser->isAnonymous() || !in_array($routeName, $restRouteNames, TRUE)) {
+      return;
+    }
+
+    $webformId = $this->routeMatch->getParameter('webform_id');
+    $submissionUuid = $this->routeMatch->getParameter('uuid');
+
+    // Handle webform submission.
+    if ('rest.webform_rest_submit.POST' === $routeName) {
+      try {
+        $content = json_decode($event->getRequest()->getContent(), TRUE, 512, JSON_THROW_ON_ERROR);
+        $webformId = (string) $content['webform_id'];
+      }
+      catch (\JsonException $exception) {
```

silently ignoring the errors is not a recommended approach.
If that is an error we can/should ignore, please add a clarifying comment for developers

---

Cleaned up in https://github.com/OS2Forms/os2forms_rest_api/pull/1/commits/480341547ccf8e05816a41c1f283e37abbdfde6d.